libr/core/cmd_search.c

```diff
@@ -2422,23 +2428,21 @@ static void do_string_search(RCore *core, RInterval search_itv, struct search_pa
 					}
 					(void)r_io_read_at (core->io, at, buf, len);
 				}
-				if (param->crypto_search) {
-					// TODO support backward search
-					int t = 0;
-					if (param->aes_search) {
-						t = r_search_aes_update (core->search, at, buf, len);
-					} else if (param->privkey_search) {
-						t = r_search_privkey_update (core->search, at, buf, len);
-					}
-					if (!t || t > 1) {
-						break;
+				r_search_update (core->search, at, buf, len);
+				if (param->aes_search) {
+					// Adjust length to search between blocks.
+					if (len == core->blocksize) {
```

Comment:
why is this needed? It is not clear to me, sorry. If the code is generic enough, there should be no need to update the len here in different ways based on the kind of search. How it it done for other cases? Can we do the same thing?

Reply:
It is needed to search a key even when it is spread over two blocks (defined by the `b` command). I tried to figure out for the other cases how it is handle but I could not understand clearly and I was afraid to break them.